src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs

```diff
@@ -191,12 +192,35 @@ public void Write(JsonWriter writer)
                 writer.WritePropertyName("tags");
                 writer.WriteStartObject();
 
+                // this will be used when we convert int, double, int[], double[] to string
+                var originalUICulture = Thread.CurrentThread.CurrentUICulture;
+                Thread.CurrentThread.CurrentUICulture = CultureInfo.InvariantCulture;
+
                 foreach (var tag in this.Tags.Value)
                 {
                     writer.WritePropertyName(tag.Key);
-                    writer.WriteValue(tag.Value);
+                    switch (tag.Value)
+                    {
+                        case string stringVal:
+                            writer.WriteValue(stringVal);
+                            break;
+                        case int[] intArrayValue:
+                            writer.WriteValue(string.Join(",", intArrayValue));
+                            break;
+                        case double[] doubleArrayValue:
+                            writer.WriteValue(string.Join(",", doubleArrayValue));
+                            break;
+                        case bool[] boolArrayValue:
+                            writer.WriteValue(string.Join(",", boolArrayValue));
+                            break;
+                        default:
+                            writer.WriteValue(tag.Value.ToString());
+                            break;
+                    }
                 }
 
+                Thread.CurrentThread.CurrentUICulture = originalUICulture;
```

Comment:
Should we be defensive here? If the above code fails for some reason we may end up not setting the `CurrentUICulture` back to the previous value.

Reply:
sure. i can add a Try/Finally in both cases

Reply:
done.